From a4fec4dd6a2ff2209047ea9931cbf50bb4521819 Mon Sep 17 00:00:00 2001
From: Lepton Wu <lepton@chromium.org>
Date: Sun, 15 Sep 2019 20:05:44 -0700
Subject: [PATCH 1/1] virgl: Remove formats with unusual sample count.

Most GPU require the sample count is power of 2. Just remove those
formats with unusual sample count. This decreases dEQP EGL tests run
time a lot.

Signed-off-by: Lepton Wu <lepton@chromium.org>
Reviewed-by: Eric Anholt <eric@anholt.net>
---
 src/gallium/drivers/virgl/virgl_screen.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gallium/drivers/virgl/virgl_screen.c b/src/gallium/drivers/virgl/virgl_screen.c
index 13e5bf57d07a..bfa42b591029 100644
--- a/src/gallium/drivers/virgl/virgl_screen.c
+++ b/src/gallium/drivers/virgl/virgl_screen.c
@@ -659,6 +659,9 @@ virgl_is_format_supported( struct pipe_screen *screen,
    if (MAX2(1, sample_count) != MAX2(1, storage_sample_count))
       return false;
 
+   if (!util_is_power_of_two_or_zero(sample_count))
+      return false;
+
    assert(target == PIPE_BUFFER ||
           target == PIPE_TEXTURE_1D ||
           target == PIPE_TEXTURE_1D_ARRAY ||
-- 
2.24.0.rc0.303.g954a862665-goog

